### DISCUSSION TITLE
Are you supposed to include environment files in dockerignore file?

### DISCUSSION THREAD

**Discussion by IDrumsey** [vercel/next.js#53915] [Help]
### Summary

With the .env.local file, are you supposed to ignore that from docker? Also, as far as I'm aware, the other env files like .env.production aren't supposed to have sensitive keys, so are those different and don't need to be ignored?

**Comment by devjiwonchoi** [comment]
Yes, it is recommended to add `.env*.local` on both `.gitignore` and `.dockerignore`.

### Dotenv Files

They are based on the `NODE_ENV`; `test | development | production`.

`.env.$(NODE_ENV)` files such as `.env.production` are meant to be for the development process running with `next start`.
So yes, don't let them have secrets for the actual production, but keep them in `.env.production.local` if you need to.

### Reference
- [Environment Variable Load Order](https://nextjs.org/docs/app/building-your-application/configuring/environment-variables#environment-variable-load-order)

**Reply by lermontex** [reply]
@devjiwonchoi, but in this case, during local development, these files will not get into the Docker container.

In this regard, the user will be forced to use files like` .env.development` and `env.production`, which may lead to sending files to the GIT repository.

What do you think about this?
